flash/image/classification/integrations/baal/dropout.py

```diff
@@ -31,10 +33,24 @@ def __init__(self, module: flash.Task, inference_iteration: int):
             raise MisconfigurationException("The model should contain at least 1 dropout layer.")
         self.inference_iteration = inference_iteration
 
+    @property
+    def data_pipeline(self) -> DataPipeline:
+        return super().data_pipeline
+
+    @data_pipeline.setter
+    def data_pipeline(self, data_pipeline: DataPipeline):
+        self.parent_module.data_pipeline = data_pipeline
+
     def predict_step(self, batch, batch_idx, dataloader_idx: int = 0):
         out = []
         for _ in range(self.inference_iteration):
             out.append(self.parent_module.predict_step(batch, batch_idx, dataloader_idx=dataloader_idx))
 
         # BaaL expects a shape [num_samples, num_classes, num_iterations]
         return torch.tensor(out).permute((1, 2, 0))
+
+    def on_predict_dataloader(self) -> None:
+        if self.parent_module.data_pipeline is not None:
+            self.parent_module.data_pipeline._detach_from_model(self.parent_module, RunningStage.PREDICTING)
+            self.parent_module.data_pipeline._attach_to_model(self.parent_module, RunningStage.PREDICTING)
```

Comment:
Could you explain what's happening here?

Reply:
The predict_step is being automatically wrap with the postprocessor and serializer. Without overriding the function, it would wrap the InferenceModel and not its children.

Reply:
This enable self.parent_model.predict_step to actually returns the Probabilities.

Reply:
Could you explain in a comment in the code @tchaton? it looks as if you're detaching the predict pipeline then re-attaching it straight after with no changes whatsoever.